### DISCUSSION TITLE
Overriding default grid gap in Fieldsets and Sections

### DISCUSSION THREAD

**Discussion by kokoshneta** [filamentphp/filament#6032] [Help]
When adding a Section or Fieldset in a Filament Form, the default grid gap in the content area is set to 1.5 rem by the Tailwind class `gap-6`. This is hard-coded on line 8 of [`/forms/resources/views/component-container.blade.php`](https://github.com/filamentphp/filament/blob/2.x/packages/forms/resources/views/component-container.blade.php):

```blade
<x-filament-support::grid
  :default="$getColumns('default')"
  :sm="$getColumns('sm')"
  :md="$getColumns('md')"
  :lg="$getColumns('lg')"
  :xl="$getColumns('xl')"
  :two-xl="$getColumns('2xl')"
  class="filament-forms-component-container gap-6"
>
```

Passing along a different gap class using `->extraAttributes()` doesn’t work, since that will be applied to the second-level container element `.filament-forms-{section/fieldset}-component` (just below the `.col-span-full` wrapper used with all Sections and Fieldsets), which is two levels above the component container `.filament-forms-component-container` where the gap is defined. For example, if you pass `->extraAttributes(['class' => 'gap-16'])`, this is the result:

```html
<div class="col-span-full">
  <div class="filament-forms-section-component rounded-xl border border-gray-300 bg-white **gap-16**"> <!-- Inserted here -->
    <div class="filament-forms-section-header-wrapper flex rtl:space-x-reverse overflow-hidden rounded-t-xl min-h-[56px] px-4 py-2 items-center bg-gray-100">
      <div class="filament-forms-section-header flex-1 space-y-1">
        <h3 class="font-bold tracking-tight pointer-events-none text-xl font-bold">
          Header
        </h3>
      </div>
    </div>

    <div class="filament-forms-section-content-wrapper">
      <div class="filament-forms-section-content p-6">
        <div class="grid grid-cols-1 lg:grid-cols-3 filament-forms-component-container **gap-6**"> <!-- Needed here-->
          Content here
        </div>
      </div>
    </div>
  </div>
</div>
```

Is there some way to override this value, short of creating a completely custom layout?

**Comment by danharrin** [comment]
There is not a way to customize the gap, no. I think that is probably a good thing, as I don't want people to deviate from our default design system, which was built in a way to make things look consistent.

Writing a custom view for this component is exactly what you should do, as then you own that code and have full responsibility to make sure it looks good in your app.

**Reply by kokoshneta** [reply]
The issue is when you have a layout block that contains a grid of Rich Editors or Markdown Editors – they sort of flow into each other and become hard to distinguish visually. Their toolbars sit outside the editor pane and take up extra space, so the labels actually end up being closer to the editor _above_ than to the one they belong to.

But if there’s no built-in way to change the gap (or the margin of the editor’s container), I think I’ll just live with it. It’s too much bother to write and maintain separate copies of the Section and Fieldset (and perhaps other) components just for that.

**Reply by shomarim** [reply]
You can write custom css to override the gap. I wanted tighter spacing so I changed gap-6 for any child of a fi-section-content. Modify your `resources/css/app.css` file.
```
.fi-section-content .gap-6 {
    gap: 0.5rem;
}

.fi-section-content .grid.gap-y-2 {
    gap: 0;
}
```

**Reply by anklesh-chauhan** [reply]
> You can write custom css to override the gap. I wanted tighter spacing so I changed gap-6 for any child of a fi-section-content. Modify your `resources/css/app.css` file.
> 
> ```
> .fi-section-content .gap-6 {
>     gap: 0.5rem;
> }
> 
> .fi-section-content .grid.gap-y-2 {
>     gap: 0;
> }
> ```

This has resolve my issue. Thanks.
